src/burnchains/bitcoin/mod.rs

```diff
@@ -162,3 +171,51 @@ pub enum BitcoinNetworkType {
     Testnet,
     Regtest
 }
+
+#[derive(Debug, PartialEq, Clone, Eq, Serialize, Deserialize)]
+pub struct BitcoinTxOutput {
+    pub address: BitcoinAddress,
+    pub units: u64
+}
+
+#[derive(Debug, PartialEq, Clone, Eq, Serialize, Deserialize)]
+pub enum BitcoinInputType {
+    Standard,
+    SegwitP2SH
+}
+
+#[derive(Debug, PartialEq, Eq, Clone, Serialize, Deserialize)]
+pub struct BitcoinTxInput {
+    pub keys: Vec<BitcoinPublicKey>,
+    pub num_required: usize,
+    pub in_type: BitcoinInputType
+}
+
+#[derive(Debug, PartialEq, Clone, Serialize, Deserialize)]
+pub struct BitcoinTransaction {
+    pub txid: Txid,
+    pub vtxindex: u32,
+    pub opcode: u8,
+    pub data: Vec<u8>,
+    pub inputs: Vec<BitcoinTxInput>,
+    pub outputs: Vec<BitcoinTxOutput>
+}
+
+#[derive(Debug, PartialEq, Clone, Serialize, Deserialize)]
+pub struct BitcoinBlock {
+    pub block_height: u64,
+    pub block_hash: BurnchainHeaderHash,
+    pub parent_block_hash: BurnchainHeaderHash,
+    pub txs: Vec<BitcoinTransaction>,
+}
+
+impl BitcoinBlock {
+    pub fn new(height: u64, hash: &BurnchainHeaderHash, parent: &BurnchainHeaderHash, txs: &Vec<BitcoinTransaction>) -> BitcoinBlock {
+        BitcoinBlock {
+            block_height: height,
+            block_hash: hash.clone(),
+            parent_block_hash: parent.clone(),
+            txs: txs.clone(),
```

Comment:
I'm confused by this function and the way it's being used:
Arguments `hash` and `parent` are instances of `BurnchainHeaderHash`, and hydrating `block_hash` and `parent_block_hash` (header_hash -> block_hash).

Reply:
In most blockchains, the hash of the block header and the hash of the block itself refer to the same hash, so the code here uses "header hash" and "block hash" interchangeably.